### PR TITLE
Add page about citing software

### PR DESCRIPTION
**Pull request by finsberg**
Add documentation about how to cite software

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe):

Fix https://github.com/scientificcomputing/scientificcomputing.github.io/issues/41

### CODE REVIEW

File: .cspell_dict.txt [hunk 1]
```diff
@@ -11,14 +11,17 @@ Daversin
 docname
 docnames
 Dokken
+DOLFI
```
Comment (jorgensd) [inline .cspell_dict.txt:14]: Do we really need `DOLFI` as a word?

**Review by jorgensd — APPROVED** [review]

File: .cspell_dict.txt (hunk 1 above)
Comment (finsberg) [reply .cspell_dict.txt:14]: Not sure how that happend. I agree, we should remove this.

Comment (finsberg) [inline .cspell_dict.txt:14]:
```suggestion
```

**Comment by jorgensd** [comment]
I guess we need to leave it in there for some strange reason xD

**Comment by finsberg** [comment]
> I guess we need to leave it in there for some strange reason xD

I think it should work now. Had to add dolfinx first ;)

**Pull request merged**